pkg/controller/controller.go

```diff
@@ -36,6 +38,7 @@ type Controller struct {
 	pullPolicy               string // Options: IfNotPresent, Always, or Never
 	verbose                  string // verbose levels: 1, 2, ...
 	podExpectations          *expectations.UIDTrackingControllerExpectations
+	ownerController          common.ControllerType
```

Comment:
So maybe just something like:
var (ImportController = "importer",
       CloneController = "cloner",
       UploadController = "upload",
       ...)
Then just add:
    "controllerType: string"
and pass that in when calling NewController.

If we're really concerned about the variable then it can be a constant, that's fine.  The reason this matters though IMO is if we don't need 3 locations and 3 data structures to handle this in multiple files we should try and keep it as simple as we can (ie stuff it all into the controller package, that's the only place it should be appropriate for the most part).